src/TaskManager.php

```diff
@@ -286,6 +292,38 @@ public function filterByContext($context)
     }
 
     /**
+     * @param string $uuid
+     * @return Task
+     */
+    public function getReference($uuid)
+    {
+        if (isset($this->tasks[$uuid])) {
+            return $this->tasks[$uuid];
+        }
+
+        $self    = $this;
+        $factory = new LazyLoadingValueHolderFactory();
+
+        $initializer = function (
```

Comment:
I would be interested in getting this segment explained, when you have the time. Looks really ... freaky.